File: services/typescript/fastify/src/routes/handlers/invalid_path.ts
import type { RouteHandler } from "fastify";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "../../logger.js";

export const invalidPathHandler: RouteHandler = (request, reply) => {
  getLogger().warn(
    { method: request.method, path: request.url },
    "Invalid path"
  );
  sendError(reply, AppErrors.InvalidPath);
};

File: services/python/fastapi/src/routes/handlers/invalid_path.py
import structlog
from fastapi import Request

from routes.errors import AppErrors, send_error


async def invalid_path_handler(request: Request, exc: Exception):
    structlog.get_logger().warn(
        "Invalid path", method=request.method, path=request.url.path
    )

    return send_error(AppErrors.InvalidPath)
